dist/tools/openocd/openocd.sh

```diff
@@ -421,7 +432,13 @@ do_debugserver() {
             -c 'init' \
             ${OPENOCD_DBG_EXTRA_CMD} \
             -c 'targets' \
-            -c 'halt'"
+            -c 'halt' & \
+            echo \$! > $OCD_PIDFILE ; \
+            wait \$(cat $OCD_PIDFILE)" &
```

Comment:
Why is the background `wait` needed?
Does it keep the script (and trap) alive for the cleanup?

Reply:
I just tested it without the ampersand in L437 and I think what happens then is that the shell waits for the openocd process to finish, while it is running obviously. So the script does not get to the `while read` until something on the remote side kills openocd